classwork/04-classwork.qmd

```diff
@@ -13,57 +13,57 @@ Setup from deck 3
 
 ```{r}
 library(tidymodels)
+library(modeldatatoo)
 
-data("tree_frogs", package = "stacks")
+taxi <- data_taxi()
 
-tree_frogs <- tree_frogs %>%
-  mutate(t_o_d = factor(t_o_d),
-         age = age / 86400) %>% 
-  filter(!is.na(latency)) %>%
-  select(-c(clutch, hatched))
+taxi <- taxi %>%
+  mutate(month = factor(month, levels = c("Jan", "Feb", "Mar", "Apr"))) %>% 
+  select(-c(id, duration, fare, tolls, extras, total_cost, payment_type)) %>% 
+  drop_na()
 
 set.seed(123)
 
-frog_split <- initial_split(tree_frogs, prop = 0.8, strata = latency)
-frog_train <- training(frog_split)
-frog_test <- testing(frog_split)
+taxi_split <- initial_split(taxi, prop = 0.8, strata = tip)
+taxi_train <- training(taxi_split)
+taxi_test <- testing(taxi_split)
 
-tree_spec <- decision_tree(cost_complexity = 0.001, mode = "regression")
-tree_wflow <- workflow(latency ~ ., tree_spec)
-tree_fit <- fit(tree_wflow, frog_train)
+tree_spec <- decision_tree(mode = "classification")
+tree_wflow <- workflow(tip ~ ., tree_spec)
+tree_fit <- fit(tree_wflow, taxi_train)
 ```
 
 ## Metrics for model performance
 
 `metrics()` returns a standard set of metrics
 
 ```{r}
-augment(tree_fit, new_data = frog_test) %>%
-  metrics(latency, .pred)
+augment(tree_fit, new_data = taxi_test) %>%
+  metrics(tip, .pred_class, .pred_yes)
```

Comment:
show hard and soft class metrics or just one?

Reply:
I think we need to show both eventually.

I think we should avoid `metrics()` altogether

Reply:
I'm game! Do what you think is good in the updates to deck 4 and then we'll update the classwork accordingly